frontend/src/metabase/components/EntityMenu.jsx

```diff
@@ -25,17 +25,29 @@ class EntityMenu extends Component {
     props: Props
 
     state = {
-        open: false
+        open: false,
+        freezeMenu: false,
+        menuItemContent: null
     }
 
     toggleMenu = () => {
+        if (this.state.freezeMenu) return;
+
         const open = !this.state.open
-        this.setState({ open })
+        this.setState({ open, menuItemContent: null })
+    }
+
+    setFreezeMenu = (freezeMenu) => {
+        this.setState({ freezeMenu })
+    }
+
+    replaceMenuWithItemContent = (menuItemContent) => {
+       this.setState({ menuItemContent })
     }
 
     render () {
         const { items, triggerIcon }  = this.props
-        const { open } = this.state
+        const { open, menuItemContent } = this.state
```

Comment:
It'd be nice to update `EntityMenu.info.js` with examples of the ability to freeze and have new content to keep things documented.